app/code/community/Bolt/Boltpay/Model/Observer.php

```diff
@@ -291,4 +291,74 @@ public function validateBeforeOrderCommit($observer) {
         $orderModel = Mage::getModel('boltpay/order');
         $orderModel->validateBeforeOrderCommit($observer);
     }
+
+    /**
+     * Automatically create an invoice after creating a shipment
+     *
+     * event: sales_model_service_quote_submit_before
+     *
+     * @param Varien_Event_Observer $observer Observer event contains a shipment object
+     */
+    public function createInvoiceAfterCreatingShipment($observer)
+    {
+        $shipment = $observer->getEvent()->getShipment();
+        $order = $shipment->getOrder();
+        if (
```

Comment:
Just clarifying this is ensuring we haven't already invoiced this and that we are only dealing with Bolt orders correct

Reply:
@ryan-timothy-albert Yes, that's correct.